addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1064 @@
+            'count': False,
+            'end_type': False,
+            'mo': False,
+            'tu': False,
+            'we': False,
+            'th': False,
+            'fr': False,
+            'sa': False,
+            'su': False,
+            'day': False,
+            'week_list': False
+        }
+
+    def _parse_rrule(self, rule, data, date_start):
+        day_list = ['mo', 'tu', 'we', 'th', 'fr', 'sa', 'su']
+        rrule_type = ['yearly', 'monthly', 'weekly', 'daily']
+        r = rrule.rrulestr(rule, dtstart=fields.Datetime.from_string(date_start))
+
+        if r._freq > 0 and r._freq < 4:
+            data['rrule_type'] = rrule_type[r._freq]
+        data['count'] = r._count
+        data['interval'] = r._interval
+        data['final_date'] = fields.Datetime.to_string(r._until)
+        #repeat weekly
+        if r._byweekday:
+            for i in xrange(0, 7):
+                if i in r._byweekday:
+                    data[day_list[i]] = True
+            data['rrule_type'] = 'weekly'
+        #repeat monthly by nweekday ((weekday, weeknumber), )
+        if r._bynweekday:
+            data['week_list'] = day_list[r._bynweekday[0][0]].upper()
+            data['byday'] = str(r._bynweekday[0][1])
+            data['month_by'] = 'day'
+            data['rrule_type'] = 'monthly'
+
+        if r._bymonthday:
+            data['day'] = r._bymonthday[0]
+            data['month_by'] = 'date'
+            data['rrule_type'] = 'monthly'
+
+        #repeat yearly but for openerp it's monthly, take same information as monthly but interval is 12 times
+        if r._bymonth:
+            data['interval'] = data['interval'] * 12
+
+        #FIXEME handle forever case
+        #end of recurrence
+        #in case of repeat for ever that we do not support right now
+        if not (data.get('count') or data.get('final_date')):
+            data['count'] = 100
+        if data.get('count'):
+            data['end_type'] = 'count'
+        else:
+            data['end_type'] = 'end_date'
+        return data
+
+    @api.multi
+    def _track_subtype(self, init_values):
+        record = self[0]
+        if 'start' in init_values and record.start:
+            return 'calendar.subtype_invitation'
+        elif 'location' in init_values and record.location:
+            return 'calendar.subtype_invitation'
+        return super(CalendarEvent, self)._track_subtype(init_values)
+
+    @api.onchange('partner_ids')
+    def onchange_partner_ids(self):
+        """ The basic purpose of this method is to check that destination partners
+            effectively have email addresses. Otherwise a warning is thrown.
+            :param value: value format: [[6, 0, [3, 4]]]
+        """
+        field_values = self._fields['partner_ids'].convert_to_onchange(self.partner_ids)
+        if not field_values or not field_values[0] or not field_values[0][0] == 6:
+            return
+
+        self.partner_ids = self.check_partners_email()
```

Comment:
hunh, it will assign warning in partner_ids if check_partners_email returns warning